src/submission/models.py

```diff
@@ -298,6 +299,30 @@ def get_queryset(self):
         return super(PreprintManager, self).get_queryset().filter(is_preprint=True)
 
 
+class DynamicChoiceField(models.CharField):
+    def __init__(self, dynamic_choices=(), *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dynamic_choices = dynamic_choices
+
+    def formfield(self, *args, **kwargs):
+        form_element = super().formfield(**kwargs)
+        for choice in self.dynamic_choices:
+            form_element.choices.append(choice)
+        return form_element
+
+    def validate(self, value, model_instance):
+        """
+        Validates value and throws ValidationError.
+        """
+        try:
+            super().validate(value, model_instance)
+        except exceptions.ValidationError:
+            # Check if the value is in dynamic choices and remove the
+            # error message if it is
+            if 'invalid_choice' in self.error_messages and value in self.dynamic_choices:
```

Comment:
Does this work? I think `value` is a string whilst `self.dynamic_choices` is a tuple of tuples